version: "3"
services:
  plex:
    image: linuxserver/plex
    environment:
      - TZ=${PLEX_TZ}
      - PLEX_CLAIM=${PLEX_CLAIM}
      - PUID=${PLEX_MEDIA_LIBRARY_OWNER_USER_ID}
      - PGID=${PLEX_MEDIA_LIBRARY_OWNER_GROUP_ID}
      - VERSION=docker
    networks:
      - traefik
    volumes:
      - ${PLEX_LIBRARY_FOLDER}:/config
      - ${PLEX_MEDIA_FOLDER}:/data
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      labels:
        - traefik.enable=true
        - traefik.http.routers.plex.rule=Host(`plex.${DOMAIN}`)
        - traefik.http.routers.plex.entryPoints=https
        - traefik.http.routers.plex.tls=true
        - traefik.http.routers.plex.tls.certresolver=letsencrypt
        - traefik.http.services.plex.loadbalancer.server.port=32400

networks:
  traefik:
    external: true
